server/src/main/java/com/hedera/block/server/persistence/storage/compression/ZstdCompression.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.persistence.storage.compression;
+
+import com.github.luben.zstd.ZstdOutputStream;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * TODO: add documentation
+ */
+public class ZstdCompression implements Compression {
+    private static final String ZSTD_EXTENSION = ".zstd";
+
+    @NonNull
+    @Override
+    public OutputStream newCompressingOutputStream(@NonNull final Path pathToFile) throws IOException {
+        final Path localPathToFile = pathToFile.resolveSibling(pathToFile.getFileName() + ZSTD_EXTENSION);
+        return new ZstdOutputStream(Files.newOutputStream(localPathToFile));
```

Comment:
The `zstd` lib allows us to set a custom `compression level` for the out stream, do we need anything other than their default?

Reply:
I believe we wanted this to be configurable, with default of `6`.  Min is `0` and Max is `9`, if I recall correctly.

Reply:
Thanks, that makes sense, I will look into that.

Reply:
@jsync-swirlds the java zstd lib does not provide the best documentation, but after running the following:

```java
public static void main(String[] args) {
    System.out.println(Zstd.minCompressionLevel());
    System.out.println(Zstd.defaultCompressionLevel());
    System.out.println(Zstd.maxCompressionLevel());
}
```

I get this as a result:

```text
-131072
3
22
```

I suppose I _should_ trust these values... document the range and have a precondition check for the provided value externally, this best fails before we have a running app...

Reply:
I think we should add the link to the documentation on the comments of the configurable property and try out with 6.

Reply:
@jsync-swirlds @AlfredoG87 I added the configuration for the compression level, left a `todo` there since I am not certain what should be the final version for the default (what it should be based on) and if we have defined a range internally that should be used, otherwise maybe we should use the lib's ranges by invoking it's methods.

Reply:
My recollection of min/max is very likely faulty.  We should use what is documented for the library.
We'll have to test and optimize a bit regardless, so it doesn't matter much what we start with as long as it is not at the extremes.

Reply:
✅ Done. I am using the lib's defaults for now.